commit aae40dd018302eae5517647c6083efc37b0938e6
Author: Stefan Wahren <stefan.wahren@i2se.com>
Date:   Sat Oct 13 20:51:23 2018 +0200

    staging: vchiq_arm: Fix platform device unregistration
    
    In error case platform_device_register_data would return an ERR_PTR
    instead of NULL. So we better check this before unregistration.
    
    Fixes: 37b7b3087a2f ("staging/vc04_services: Register a platform device for the camera driver.")
    Signed-off-by: Stefan Wahren <stefan.wahren@i2se.com>

diff --git a/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c b/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c
index fe431302a030..7ab3c29c03a5 100644
--- a/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c
+++ b/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_arm.c
@@ -3656,7 +3656,8 @@ static int vchiq_probe(struct platform_device *pdev)
 
 static int vchiq_remove(struct platform_device *pdev)
 {
-	platform_device_unregister(bcm2835_camera);
+	if (!IS_ERR(bcm2835_camera))
+		platform_device_unregister(bcm2835_camera);
 	vchiq_debugfs_deinit();
 	device_destroy(vchiq_class, vchiq_devid);
 	class_destroy(vchiq_class);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1230_aae40dd018302eae5517647c6083efc37b0938e6.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1230_aae40dd018302eae5517647c6083efc37b0938e6.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
